### PR TITLE
sessionctx: enable IndexMerge by default

### LINKED ISSUE #29597: enable tidb_enable_index_merge by default
## Enhancement
Many application query will use sql as below format, although there are both c1/c2 are indexed by separate index.
Without enabling indexMerge, the query is forced full table scan.
I would like to propose to enable `tidb_enable_index_merge` by default.
```
select * from t where c1 = 'abc' or c2 = 'xyz';
index1 on (c1)
index2 on (c2)
```

### PR DESCRIPTION
**Pull request by guo-shaoge**
Signed-off-by: guo-shaoge <shaoge1994@163.com>

### What problem does this PR solve?

Issue Number: close #29597

Problem Summary:  Enable IndexMerge by default. It's part of IndexMerge GA Plan.

### What is changed and how it works?
`tidb_enable_index_merge` is introduced in v4.0.0(`Off` by default), and it doesn't exist in 3.x. So
1. when a 3.x cluster upgrade to v5.4+, we want it to be `Off`.
2. when a 4.x/5.x cluster upgrade to v5.4+, we want it to be the original value(user may set it to `On`).
3. when a new v5.4+ cluster is created, we want it to be `On`.

So During upgrade, we check if `tidb_enable_index_merge` exists in `mysql.GLOBAL_VARIABLES`. If it exists, we do nothing. Otherwise insert it and set its value as `Off`.

### Check List

Tests 

- [ ] Unit test
- [x] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [x] Affects user behaviors
- [ ] Contains syntax changes
- [x] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

```release-note
sessionctx: enable IndexMerge by default
```

### CODE REVIEW

File: session/bootstrap.go [hunk 1]
```diff
@@ -541,11 +541,14 @@ const (
 	// version80 fixes the issue https://github.com/pingcap/tidb/issues/25422.
 	// If the TiDB upgrading from the 4.x to a newer version, we keep the tidb_analyze_version to 1.
 	version80 = 80
+	// version80 insert "tidb_enable_index_merge|off" to mysql.GLOBAL_VARIABLES if there is no tidb_enable_index_merge.
```
Comment (guo-shaoge) [inline session/bootstrap.go:544]:
```suggestion
	// version81 insert "tidb_enable_index_merge|off" to mysql.GLOBAL_VARIABLES if there is no tidb_enable_index_merge.
```

File: session/bootstrap_test.go [hunk 2]
```diff
@@ -928,3 +928,118 @@ func TestAnalyzeVersionUpgradeFrom300To500(t *testing.T) {
 	require.Equal(t, 1, row.Len())
 	require.Equal(t, "1", row.GetString(0))
 }
+
+func TestIndexMergeUpgradeFrom300To500(t *testing.T) {
```
Comment (XuHuaiyu) [inline session/bootstrap_test.go:932]: Add a test for `notBootstrap` to `540`

Comment (guo-shaoge) [reply session/bootstrap_test.go]: Added in `TestIndexMergeInNewCluster()` .

**Review by XuHuaiyu — APPROVED** [review]
LGTM

**Review by morgo — APPROVED** [review]

File: session/bootstrap.go [hunk 3]
```diff
@@ -1630,6 +1634,27 @@ func upgradeToVer79(s Session, ver int64) {
...
+	terror.MustNil(err)
+	req := rs.NewChunk(nil)
+	err = rs.Next(ctx, req)
+	terror.MustNil(err)
+	if req.NumRows() != 0 {
+		return
+	}
+
+	mustExecute(s, "INSERT HIGH_PRIORITY IGNORE INTO %n.%n VALUES (%?, %?);",
+		mysql.SystemDB, mysql.GlobalVariablesTable, variable.TiDBEnableIndexMerge, variable.Off)
```
[the remark this answers is not in the document]
Comment (guo-shaoge) [reply session/bootstrap.go:1682]: Yes. That's the behavior I want. (when user upgrades a pre-4.0 version TiDB to 5.4+, tidb_enable_index_merge should be off by default.)

And I will add this comment. Thanks for reviewing.

**Comment by guo-shaoge** [comment]
Could you please /merge this pr. I am not a commiter. Thanks! @morgo

**Pull request merged**